OTP doesn't build when --without-erl_interface is given

**Describe the bug**
OTP doesn't build when `--without-erl_interface` is given. Reproduced when building a Docker image and in an Ubuntu VM.

**To Reproduce**
```
$ git clone https://github.com/erlang/otp
$ cd otp
$ ./otp_build autoconf
$ ./configure --without-erl_interface
...

*********************************************************************
**********************  APPLICATIONS DISABLED  **********************
*********************************************************************

jinterface     : No Java compiler found
odbc           : ODBC library - link check failed
erl_interface  : User gave --without-erl_interface option

*********************************************************************
*********************************************************************
**********************  APPLICATIONS INFORMATION  *******************
*********************************************************************

wx             : wxWidgets was not compiled with --enable-webview or wxWebView developer package is not installed, wxWebView will NOT be available

*********************************************************************
*********************************************************************
**********************  DOCUMENTATION INFORMATION  ******************
*********************************************************************

documentation  : 
                 xsltproc is missing.
                 fop is missing.
                 xmllint is missing.
                 The documentation cannot be built.

*********************************************************************
$ make -j 8
Makefile:1220: warning: overriding recipe for target 'dialyzer'
Makefile:510: warning: ignoring old recipe for target 'dialyzer'
 MAKE	depend
make[1]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
 MAKE	generate
make[2]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
 GEN	x86_64-unknown-linux-gnu/opt/jit/OPCODES-GENERATED
 GEN	x86_64-unknown-linux-gnu/opt/jit/TABLES-GENERATED
 GEN	x86_64-unknown-linux-gnu/opt/jit/erl_alloc_types.h
 GEN	x86_64-unknown-linux-gnu/erl_version.h
 GEN	x86_64-unknown-linux-gnu/opt/jit/driver_tab.c
make[3]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
 MAKE	opt
make[4]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
 GEN	x86_64-unknown-linux-gnu/preload.c
 GEN	obj/x86_64-unknown-linux-gnu/opt/depend.mk
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/simple_c_gc.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/re.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_lexer.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_main.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_node.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_parser.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_printers.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_string.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_symbol.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_utils.o
 CC	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/x86_64-unknown-linux-gnu/ycf_yield_fun.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/ethr_aux.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/ethr_atomics.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/ethr_mutex.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/ethr_cbf.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/ethread.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/ethr_event.o
 CC	obj/x86_64-unknown-linux-gnu/opt/erl_memory_trace_parser.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/erl_memory_trace_parser.o
 CC	obj/x86_64-unknown-linux-gnu/opt/erl_printf_format.o
 CC	obj/x86_64-unknown-linux-gnu/opt/erl_printf.o
 CC	obj/x86_64-unknown-linux-gnu/opt/erl_misc_utils.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/erl_printf_format.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/erl_printf.o
 CC	obj/x86_64-unknown-linux-gnu/opt/r/erl_misc_utils.o
 AR	../lib/internal/x86_64-unknown-linux-gnu/liberts_internal.a
 RANLIB	../lib/internal/x86_64-unknown-linux-gnu/liberts_internal.a
 AR	../lib/internal/x86_64-unknown-linux-gnu/libethread.a
 AR	../lib/x86_64-unknown-linux-gnu/liberts.a
 RANLIB	../lib/internal/x86_64-unknown-linux-gnu/libethread.a
 RANLIB	../lib/x86_64-unknown-linux-gnu/liberts.a
 AR	../lib/x86_64-unknown-linux-gnu/liberts_r.a
 RANLIB	../lib/x86_64-unknown-linux-gnu/liberts_r.a
 AR	../lib/internal/x86_64-unknown-linux-gnu/liberts_internal_r.a
 RANLIB	../lib/internal/x86_64-unknown-linux-gnu/liberts_internal_r.a
 LD	/home/lhoguin/rabbitmq/otp/erts/lib_src/yielding_c_fun/bin/x86_64-unknown-linux-gnu/yielding_c_fun
 GEN	obj/x86_64-unknown-linux-gnu/opt/MADE
make[4]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
make[3]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
 GEN	x86_64-unknown-linux-gnu/erl_db_insert_list.ycf.h
 GEN	x86_64-unknown-linux-gnu/erl_map.ycf.h
 GEN	x86_64-unknown-linux-gnu/utils.ycf.h
 GEN	x86_64-unknown-linux-gnu/opt/jit/GENERATED
make[2]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
make[1]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
make[1]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
 MAKE	depend
make[2]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
 GEN	x86_64-unknown-linux-gnu/opt/jit/depend.mk
make[3]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
 MAKE	depend
make[4]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
make[4]: Nothing to be done for 'depend'.
make[4]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
make[3]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
make[2]: Nothing to be done for 'depend'.
make[2]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
make[1]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/emulator'
make[1]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
 MAKE	depend
make[2]: Entering directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
make[2]: Nothing to be done for 'depend'.
make[2]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
make[1]: Leaving directory '/home/lhoguin/rabbitmq/otp/erts/lib_src'
 MAKE	erl_interface
make[1]: Entering directory '/home/lhoguin/rabbitmq/otp/lib/erl_interface'
=== Entering application erl_interface
make[2]: Entering directory '/home/lhoguin/rabbitmq/otp/lib/erl_interface/src'
 MAKE	opt
make[3]: Entering directory '/home/lhoguin/rabbitmq/otp/lib/erl_interface/src'
make[3]: x86_64-unknown-linux-gnu/Makefile: No such file or directory
make[3]: *** No rule to make target 'x86_64-unknown-linux-gnu/Makefile'.  Stop.
make[3]: Leaving directory '/home/lhoguin/rabbitmq/otp/lib/erl_interface/src'
make[2]: *** [Makefile:31: opt] Error 2
make[2]: Leaving directory '/home/lhoguin/rabbitmq/otp/lib/erl_interface/src'
make[1]: *** [/home/lhoguin/rabbitmq/otp/make/otp_subdir.mk:29: opt] Error 2
make[1]: Leaving directory '/home/lhoguin/rabbitmq/otp/lib/erl_interface'
make: *** [Makefile:510: erl_interface] Error 2
```

**Expected behavior**
Builds.

**Affected versions**
master, 24.0-rc1

**Additional context**
Same happens when `-j 8` is not given. But slower.

--- a/configure.src
+++ b/configure.src
@@ -148,7 +148,14 @@ while test $# != 0; do
 	    case $1 in
 		--without-*)
 		    skip_app=`expr "$1" : '--without-\(.*\)'`
-		    if test -d "lib/$skip_app"; then
+                    if [ "$skip_app" = "stdlib" ] ||
+                           [ "$skip_app" = "kernel" ] ||
+                           [ "$skip_app" = "sasl" ] ||
+                           [ "$skip_app" = "compiler" ] ||
+                           [ "$skip_app" = "erl_interface" ]; then
+                        echo "ERROR: $skip_app is a mandatory application" 1>&2
+                        exit 1
+		    elif test -d "lib/$skip_app"; then
 			skip_applications="$skip_applications $skip_app"
 		    fi;;
 		*)

